### PR TITLE
ICD groundwork

### PR DESCRIPTION
**Pull request by msiglreith**
Restructuring the project into 3 crates
- `portability`: statically linkable library for native applications
- `portability-gfx`: implementation crate, required as ICD doesn't allow exposing vkFunction symbols
- `portability-icd`: ICD implementation (driver)

Mostly shifting around stuff and basics to get the driver working. Currently stuck in the `I want to create an instance'-loop

### CODE REVIEW

File: libportability-gfx/src/lib.rs [hunk 1]
```diff
@@ -5882,8 +4806,7 @@ pub struct VkWin32SurfaceCreateInfoKHR {
 impl Clone for VkWin32SurfaceCreateInfoKHR {
     fn clone(&self) -> Self { *self }
 }
-#[no_mangle]
-pub extern fn vkCreateWin32SurfaceKHR(
+pub fn gfxCreateWin32SurfaceKHR(
```
Comment (kvark) [inline libportability-gfx/src/lib.rs]: these should probably all be `#[inline]`

File: Cargo.toml [hunk 2]
```diff
@@ -1,22 +1,6 @@
...
-branch = "portable"
-
-[dependencies.gfx-backend-vulkan]
-#path = "../gfx/src/backend/vulkan"
-git = "https://github.com/kvark/gfx-rs"
-branch = "portable"
-features = ["portable"]
+[workspace]
+members = [
+    "libportability",
```
Comment (kvark) [inline Cargo.toml]: I'm not comfortable with "libportability-XXX" name. Is there any other lib that starts with "lib"?

Comment (msiglreith) [reply Cargo.toml]: Adhering to the layout of the [`rust-lang/rust`](https://github.com/rust-lang/rust/tree/master/src) repo, but I don't mind other names. suggestions?

Comment (kvark) [reply Cargo.toml]: I'm a bit confused by the fact the repo itself here is about the portability library, so having `libportability-x` inside it is redundant. This is unlike Rust repo.
However, I suppose we can end up with some binary sub-modules later on, so this is future-proof.

**Comment by kvark** [comment]
Wonderful, thank you!
bors r+

**Pull request merged**